xdsl/irdl/declarative_assembly_format_parser.py

```diff
@@ -780,23 +768,22 @@ def parse_format_directive(self) -> FormatDirective:
             return variable
         self.raise_error(f"unexpected token '{self._current_token.text}'")
 
-    def create_attr_dict_directive(self, with_keyword: bool) -> AttrDictDirective:
+    def create_attr_dict_directive(
+        self, with_keyword: bool, idx: int
+    ) -> AttrDictDirective:
         """Create an attribute dictionary directive, and update the parsing state."""
-        if self.has_attr_dict:
+        if self.attr_dict_idx != -1:
```

Comment:
Would it be possible to instead move this logic into the `parse_format` method? It feels a little more pertinent there, as the `AttrDictDirective` itself doesn't need the index, and it's more of a property of the whole format than the individual directive.

Reply:
I've moved the logic to `verify_attr_dict`